File: code/python/judgeCircle.py
class Solution:
    def judgeCircle(self, moves: str) -> bool:
        if moves == "":
            return True
        pos = [0,0]
        for i in range(len(moves)):
            if moves[i] == "U":
                pos[1] += 1
            elif moves[i] == "D":
                pos[1] -= 1
            elif moves[i] == "R":
                pos[0] += 1
            elif moves[i] == "L":
                pos[0] -= 1
        return pos[0]==0 and pos[1] == 0

File: code/java/judgeCircle.java
class Solution {
    public boolean judgeCircle(String moves) {
        if(moves.equals(""))
            return true;
        int[] pos = new int[2];
        for(int i=0;i<moves.length();i++){
            char c = moves.charAt(i);
            if(c == 'R') pos[0] += 1;
            else if(c == 'L') pos[0] -= 1;
            else if(c == 'U') pos[1] +=  1;
            else if(c == 'D') pos[1] -= 1;
        }
        return (pos[0] == 0 && pos[1] == 0);
    }
}
